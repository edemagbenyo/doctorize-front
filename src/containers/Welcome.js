import React from 'react';
import Account from '../components/Account';
import '../styles/tablets/welcome.scss'
import Doctor1 from '../images/doctor1.png';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
const Welcome = ({isLoggedIn}) => {
  console.log(isLoggedIn);
  return (
    <div className="welcome-container">
      <div className="navbar">
        <div className="menu">Menu</div>
        <div className="search">Search</div>
        <div className="account">
          <Account isLoggedIn={isLoggedIn}/>
        </div>
      </div>
      <div className="slider">
        <div className="caption">
        <h1>Welcome to doctorize</h1>
        <Link to="/">Get Started</Link>
        </div>
        <div className="slide"><img src={Doctor1} alt="Doctor"/></div>
      </div>

    </div>
  );
};

const mapStateToProps = (state)=>{
  return {
    isLoggedIn: state.auth.isLoggedIn
  }
}

export default connect(mapStateToProps,null)(Welcome);