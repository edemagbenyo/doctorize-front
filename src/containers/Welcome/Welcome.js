import React from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import Account from '../../components/Account/Account';
import './Welcome.scss';
import Doctor1 from '../../images/doctor1.png';

const Welcome = ({ isLoggedIn }) => (
  <div className="welcome-container">
    <div className="navbar">
      <div className="menu">Menu</div>
      <div className="search">Search</div>
      <div className="account">
        <Account isLoggedIn={isLoggedIn} />
      </div>
    </div>
    <div className="slider">
      <div className="caption">
        <h1>Welcome to doctorize</h1>
        <Link to="/">Get Started</Link>
      </div>
      <div className="slide"><img src={Doctor1} alt="Doctor" /></div>
    </div>

  </div>
);

Welcome.defaultProps = {
  isLoggedIn: false,
};
Welcome.propTypes = {
  isLoggedIn: PropTypes.bool,
};

const mapStateToProps = state => ({
  isLoggedIn: state.auth.isLoggedIn,
});

export default connect(mapStateToProps, null)(Welcome);
