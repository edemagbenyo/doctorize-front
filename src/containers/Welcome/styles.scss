// @media only screen and (min-width: 768px) and (max-width: 1024px) {
  a {
    text-decoration: none;
    color: #000000;
  }
  .welcome-container {
    display: flex;
    height: 100vh;
    flex-direction: column;
    // align-items: center;
    .slider {
      img {
        width: 100%;
        height: 100vh;
      }
      .caption {
        // position: absolute;
        width: 100%;
        // top: 40%;
        text-align: center;
        h1 {
          color: rgb(0, 111, 175);
          font-size: 1.9em;
        }
        a{
          background-color:rgb(0, 111, 175) ;
          color: #FFFFFF;
          font-size: 1.5em;
          padding: 10px;
        }
      }
    }
  }
  .navbar {
    display: flex;
    position: absolute;
    top: 0;
    left: 0;
    width: 100%;
    justify-content: space-between;
    .menu {
      margin: 40px;
    }
    .search {
      margin: 40px;
    }
    .account {
      margin: 40px;
      ul {
        margin: 0;
        padding: 0;
        list-style: none;
        li {
          display: inline-block;
          margin: 0 5px;
          
        }
      }
    }
  }
// }
