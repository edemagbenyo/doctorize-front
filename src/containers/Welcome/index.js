import React from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import Account from '../../components/Account';
import './styles.scss';
import Doctor1 from '../../images/doctor1.png';

const Welcome = ({ isLoggedIn }) => (
  <div className="welcome-container">
    <div className="navbar">
      <div className="menu"><h3>!Doctorize</h3></div>
      <div className="search" />
      <div className="account">
        <Account isLoggedIn={isLoggedIn} />
      </div>
    </div>
    <div className="slider">
      <div className="slide"><img src={Doctor1} alt="Doctor" /></div>
      <div className="caption">
        <h1>Welcome to doctorize</h1>
        <h2>Medical appointment at your convinience.</h2>
        <h2>Talk to a medial specialist from the comfort of your home.</h2>
        <Link to="/home">Get Started</Link>
      </div>
    </div>

  </div>
);

Welcome.defaultProps = {
  isLoggedIn: false,
};
Welcome.propTypes = {
  isLoggedIn: PropTypes.bool,
};

const mapStateToProps = state => ({
  isLoggedIn: state.auth.isLoggedIn,
});

export default connect(mapStateToProps, null)(Welcome);
