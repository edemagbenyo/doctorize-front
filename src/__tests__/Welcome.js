import React from 'react';
import { render, cleanup, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Welcome from '../containers/Welcome';


afterEach(cleanup);

describe('welcome', () => {
  it('renders', () => {
    const { asFragment } = render(<Welcome />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('contains a text', () => {
    const { debug } = render(<Welcome />);
    const text = screen.getByText('Welcome page');
    expect(text).toBeTruthy();
  });
});
