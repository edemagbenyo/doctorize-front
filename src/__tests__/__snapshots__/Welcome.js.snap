// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`welcome renders 1`] = `
<DocumentFragment>
  <div>
    <h1>
      Welcome page
    </h1>
  </div>
</DocumentFragment>
`;
